sunpy/io/fits.py

```diff
@@ -183,7 +185,20 @@ def write(fname, data, header, hdu_type=None, **kwargs):
     if not hdu_type:
         hdu_type = fits.PrimaryHDU
 
-    hdu = hdu_type(data=data, header=fits_header)
+    if isinstance(hdu_type, (fits.PrimaryHDU, fits.hdu.base.ExtensionHDU)):
+        hdu = hdu_type  # HDU already initialised
+
+        # Merge `header` into HDU's header
+        # Values in `header` take priority, including cards such as
+        # 'SIMPLE' and 'BITPIX'.
```

Comment:
I think cards like SIMPLE and BITPIX will be "corrected" to match the file requirements at write time, which is fine.